### PR TITLE
WIP named fragments - changing getFromAST

### PR DESCRIPTION
**Pull request by Poincare**
This is the first step to getting named fragments working. The getFromAST functions assume that there is only one definition in every document which is problematic when we also have FragmentDefinitions as well as OperationDefinition. This changeset fixes + tests that issue. 

Issue #80 

TODO:

- [x] Update CHANGELOG.md with your change
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

File: src/queries/getFromAST.ts [hunk 1]
```diff
@@ -4,42 +4,75 @@ import {
...
 
-  if (doc.definitions.length > 1) {
-    throw new Error('Mutation query must have exactly one operation definition.');
-  }
+export function getMutationDefinition(doc: Document): OperationDefinition {
+  checkDocument(doc);
 
-  const mutationDef = doc.definitions[0] as OperationDefinition;
+  let mutationDef: OperationDefinition = null;
+  doc.definitions.map((definition) => {
```
Comment (stubailo) [inline src/queries/getFromAST.ts]: In the case where you're not using the return value, `forEach` is usually preferred.

File: src/queries/getFromAST.ts [hunk 2]
```diff
@@ -4,42 +4,75 @@ import {
...
+    if (definition.kind === 'OperationDefinition'
+        && (definition as OperationDefinition).operation === 'mutation') {
+      mutationDef = definition as OperationDefinition;
+    }
+  });
 
-  if (mutationDef.kind !== 'OperationDefinition' || mutationDef.operation !== 'mutation') {
-    throw new Error('Must be a mutation definition.');
+  if (mutationDef === null) {
```
Comment (stubailo) [inline src/queries/getFromAST.ts]: I think `if (!mutationDef) {` is better, but up to you.

File: src/queries/getFromAST.ts [hunk 3]
```diff
@@ -4,42 +4,75 @@ import {
...
+    throw new Error('Queries must have exactly one operation definition.');
   }
+}
+
+// Returns the FragmentDefinitions from a particular document as an array
+export function getFragmentDefinitions(doc: Document): FragmentDefinition[] {
+  checkDocument(doc);
+  let fragmentDefinitions: FragmentDefinition[] = [];
+
+  doc.definitions.map((definition) => {
```
Comment (stubailo) [inline src/queries/getFromAST.ts]: In this case, you might want `filter`.

**Comment by stubailo** [comment]
Looks like a great start to me!

**Comment by Poincare** [comment]
@stubailo fixed!

**Comment by Poincare** [comment]
@stubailo fixed.

**Pull request merged**